src/commcare_cloud/fab/operations/formplayer.py

```diff
@@ -23,7 +23,8 @@ def _formplayer_jars_differ(build_dir, release_1, release_2):
     with cd(build_dir):
         with settings(warn_only=True):
             result = sudo("diff {}/libs/formplayer.jar {}/libs/formplayer.jar".format(release_1, release_2))
-    return result.return_code == 1
+    # 1 means there's a diff, 2 means one of the files doesn't exist
+    return result.return_code in (1, 2)
```

Comment:
why not `!= 0`? I believe [`diff` only returns `0` if the files are the same](https://stackoverflow.com/questions/6971284/what-are-the-error-exit-values-for-diff).

Reply:
Yeah I think the difference between `in (1, 2)` and `!= 0` has to do with what we do when there's an unexpected error code. Ideally we fail hard if anything goes wrong, return `True` if the files are positively the different and `False` if they're positively the same. I think both the way I'm doing it and modifying to `!= 0` have the same issue which is that there are plenty of ways something could error where that error would be swallowed and simply result in one of the two possible normal values (True/False).

Reply:
My thinking was that `!= 0` would be a more conservative change than `in (1, 2)`: it will cause a restart (return `True`) when it is maybe not necessary (due to some unknown failure mode of `diff`), but will always prevent a restart (return `False`) when `diff` reports that the files are the same.

As is, an unknown `diff` failure mode will prevent a (possibly necessary) restart.